public/app/screens/main/main-screen.jsx

```diff
@@ -3,35 +3,40 @@ var React = require("react");
 var Router = require('react-router');
 var RouteHandler = Router.RouteHandler;
 
-var Navigation = require('common/regions/navigation');
-var navigationStore = require('stores/navigation')
+var Navigation = require('./navigation');
+var navigationStore = require('stores/navigation');
+var StyleMixin = require('mixins/style-mixin');
 
 var style = require("./style.scss");
 
 var MainScreen = React.createClass({
-    
-    getInitialState: function() {
-        return {
-            navItems: navigationStore.getItems()
-        }
-    },
+  mixins: [StyleMixin(require('./style.scss'))],
 
-	render: function () {
-		return (
-			<div className="main-screen">
-				<div className="main-screen__navigation">
-					<Navigation items={this.state.navItems}/>
-				</div>
-				<div className="main-screen__content">
-					<RouteHandler/>
-				</div>
-			</div>
-		)
-	}
+  getInitialState: function () {
+    return {
+      navItems: navigationStore.getItems()
+    }
+  },
+
+  render: function () {
+    var navigation = <Navigation items={this.state.navItems} orientation="medium-vertical small-horizontal"/>;
+
+    return (
+      <div className="main-screen">
+        <div className="main-screen__top-row main-screen--small-only">
```

Comment:
As we have `main-screen--hidden-for-small` this could be named `main-screen--visible-for-small-only` to have a naming pendant.